### PR TITLE
[Tooltips] Fixes for dictionary objects and self-referential objects

### LINKED ISSUE #6810: Issues with ctrl+hover gesture in plot legends
#### Summary
The `ctrl+hover` gesture to get the full path has various issues.

1. When hovering over telemetry in legend, it does retrieve the full path, but  404s are in console and error notifications UI, and the hover displays `Missing: [The name of the object being hovered over]`

EDIT: This is only happening with telemetry added from the VIPER dictionary.

2. If an aggregate is dragged into a plot, `ctrl+hover` does nothing when hovering over in the plot legend.

3. When the telemetry, not overlay or other plot type, is selected in tree, The `ctrl+hover` gesture is not displaying full path.

#### Expected vs Current Behavior
No console errors. The full path is displayed without a missing object label just like hovering over the the object name in other display locations.

#### Steps to Reproduce
1. Create an overlay plot
2. Drag and drop telemetry from the VIPER dictionary into the overlay plot
3. Drag and drop an aggregate telemetry from the VIPER dictionary into the overlay plot.
4. Create a sine wave generator in the overlay plot

Issue 1
1. `Ctrl+Hover` over the VIPER telemetry from setup step 2 in the plot legend
2. Observe error notification, console error, and `missing:` prefix in hover tooltip

Issue 2
1. `Ctrl+Hover` over the aggregate telemetry from setup step 3 in the plot legend
2. Observe the tooltip does not show

Issue 3
1. `Ctrl+Hover` over the Sine Wave Generator from setup step 4 in the plot legend
2. Observe the tooltip works as expected
3. Select the Sine Wave Generator in the navigation tree
4. `Ctrl+Hover` over the Sine Wave Generator from setup step 4 in the plot legend
5. Observe the tooltip does not show 

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by khalidadil**
Closes #6810 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/api/objects/ObjectAPI.js [hunk 1]
```diff
@@ -555,27 +555,39 @@ export default class ObjectAPI {
   async getTelemetryPath(identifier, telemetryIdentifier) {
     const objectDetails = await this.get(identifier);
     const telemetryPath = [];
-    if (objectDetails.composition && !['folder'].includes(objectDetails.type)) {
-      let sourceTelemetry = objectDetails.composition[0];
+    if (['folder'].includes(objectDetails.type)) {
```
Comment (ozyx) [inline src/api/objects/ObjectAPI.js:558]: Should we change this to `if(objectDetails.type === 'folder')` ?

File: src/api/objects/ObjectAPI.js [hunk 2]
```diff
@@ -555,27 +555,39 @@ export default class ObjectAPI {
...
+      return telemetryPath;
+    }
+
+    let sourceTelemetry = null;
+    let isParentSameAsChild = true;
+    Object.keys(identifier).forEach((key) => {
```
Comment (ozyx) [inline src/api/objects/ObjectAPI.js:564]: Let's use [identifierEquals()](https://github.com/nasa/openmct/blob/f3b819a7867ab8089be7c50de54bebe6d2e0c5a5/src/api/objects/object-utils.js#L150) for this

File: src/api/objects/ObjectAPI.js [hunk 3]
```diff
@@ -555,27 +555,39 @@ export default class ObjectAPI {
...
+      }
+    });
+    if (isParentSameAsChild) {
+      sourceTelemetry = identifier;
+    } else if (objectDetails.composition) {
+      sourceTelemetry = objectDetails.composition[0];
       if (telemetryIdentifier) {
         sourceTelemetry = objectDetails.composition.find(
           (telemetrySource) =>
             this.makeKeyString(telemetrySource) === this.makeKeyString(telemetryIdentifier)
```
Comment (ozyx) [inline src/api/objects/ObjectAPI.js:576]: I think we can use `identifierEquals()` here too?

File: src/api/objects/ObjectAPI.js [hunk 4]
```diff
@@ -555,27 +555,39 @@ export default class ObjectAPI {
...
-        }
-        telemetryPath.unshift(pathObject.name);
-      });
     }
+    const compositionElement = await this.get(sourceTelemetry);
+    if (!['yamcs.telemetry', 'generator', 'yamcs.aggregate'].includes(compositionElement.type)) {
+      return telemetryPath;
+    }
+    const telemetryPathObjects = await this.getOriginalPath(compositionElement.identifier);
+    telemetryPathObjects.forEach((pathObject) => {
```
Comment (ozyx) [inline src/api/objects/ObjectAPI.js:585]: Should we just use `.map()` here instead?

**Review by ozyx — CHANGES_REQUESTED** [review]
Smoke tested this and seems to work well! I've left some suggestions mostly around increasing readability. I think we should also leave some comments throughout the function detailing what it's doing. Great work!

File: src/api/objects/ObjectAPI.js (hunk 2 above)
Comment (khalidadil) [reply src/api/objects/ObjectAPI.js:564]: ooh, good call

File: src/api/objects/ObjectAPI.js (hunk 4 above)
Comment (khalidadil) [reply src/api/objects/ObjectAPI.js:585]: yeah, I could probably do a `reverse()` and `map()` over that

File: src/api/tooltips/ToolTipAPI.js [hunk 5]
```diff
@@ -57,13 +57,22 @@ class TooltipAPI {
    * @private for platform-internal use
    */
   showTooltip(tooltip) {
+    this.removeAllTooltips();
+    this.activeToolTips.push(tooltip);
+    tooltip.show();
+  }
+
+  /**
+   * API method to allow for removing all tooltips
```
Comment (khalidadil) [inline src/api/tooltips/ToolTipAPI.js:66]: Pulled this out to allow for cleaning up tooltips where needed in the code

File: src/api/tooltips/components/tooltip-component.scss [hunk 6]
```diff
@@ -3,6 +3,7 @@
   height: auto;
   width: auto;
   padding: $interiorMargin;
+  overflow-wrap: break-word;
```
Comment (khalidadil) [inline src/api/tooltips/components/tooltip-component.scss:6]: Fixes an issue with tooltips without any spaces in them

File: src/api/objects/ObjectAPI.js [hunk 7]
```diff
@@ -554,40 +554,34 @@ export default class ObjectAPI {
...
     }
+
     const telemetryPathObjects = await this.getOriginalPath(compositionElement.identifier);
-    telemetryPathObjects.forEach((pathObject) => {
-      if (pathObject.type === 'root') {
-        return;
-      }
-      telemetryPath.unshift(pathObject.name);
-    });
+    telemetryPath = telemetryPathObjects
```
Comment (ozyx) [inline src/api/objects/ObjectAPI.js:580]: beautiful!

**Review by ozyx — APPROVED** [review]
LGTM. Great work!

**Pull request merged**